packages/block/src/header.ts

```diff
@@ -4,58 +4,154 @@ import {
   zeros,
   KECCAK256_RLP_ARRAY,
   KECCAK256_RLP,
+  rlp,
   toBuffer,
-  defineProperties,
+  unpadBuffer,
   bufferToInt,
   rlphash,
 } from 'ethereumjs-util'
-import { Blockchain, BlockHeaderData, BufferLike, BlockOptions, PrefixedHexString } from './types'
+import { Blockchain, HeaderData, BlockOptions } from './types'
 import { Buffer } from 'buffer'
 import { Block } from './block'
+import { checkBufferLength } from './util'
 
 /**
  * An object that represents the block header
  */
-export class BlockHeader {
-  public raw!: Buffer[]
-  public parentHash!: Buffer
-  public uncleHash!: Buffer
-  public coinbase!: Buffer
-  public stateRoot!: Buffer
-  public transactionsTrie!: Buffer
-  public receiptTrie!: Buffer
-  public bloom!: Buffer
-  public difficulty!: Buffer
-  public number!: Buffer
-  public gasLimit!: Buffer
-  public gasUsed!: Buffer
-  public timestamp!: Buffer
-  public extraData!: Buffer
-  public mixHash!: Buffer
-  public nonce!: Buffer
-
-  readonly _common: Common
+export class Header {
```

Comment:
I am not yet sure if I like renaming `BlockHeader` to `Header`. I don't think that `Header` would lead to any interpretation options, but `BlockHeader` is just more explicit.

Reply:
I was unsure here as well TBH, then I decided relatively quickly that this might be simpler and changed. You are likely right though so, eventually just change back.

Reply:
I have renamed it back to `BlockHeader`, for clarity.